### PR TITLE
Add an env PYTORCH_TEST_SKIP_CUDAGRAPH to skip all cuda graph-related unit tests

### PR DESCRIPTION
**Pull request by xwang233**
Skip all cuda graph-related unit tests by setting env var `PYTORCH_TEST_SKIP_CUDAGRAPH=1`

This PR refactors the `TEST_CUDA` python variable in test_cuda.py into common_utils.py. This PR also creates a new python variable `TEST_CUDA_GRAPH` in common_utils.py, which has an env var switch to turn off all cuda graph-related tests.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @bertmaher @anijain2305 @ptrblck @malfet

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -3456,7 +3452,7 @@ def test_graph_capture_simple(self):
 
         self.assertTrue(b.sum().item() == 11000.)
 
-    @unittest.skipIf((not TEST_GRAPH), "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
+    @unittest.skipIf((not TEST_CUDA_GRAPH), "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
```
Comment (malfet) [inline test/test_cuda.py:3455]: Nit
```suggestion
    @unittest.skipIf(not TEST_CUDA_GRAPH, "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
```

File: test/test_cuda.py [hunk 2]
```diff
@@ -3437,7 +3433,7 @@ def test_graph_is_current_stream_capturing(self):
                 self.assertTrue(torch.cuda.is_current_stream_capturing())
                 g.capture_end()
 
-    @unittest.skipIf((not TEST_GRAPH), "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
+    @unittest.skipIf((not TEST_CUDA_GRAPH), "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
```
Comment (malfet) [inline test/test_cuda.py:3436]:
```suggestion
    @unittest.skipIf(not TEST_CUDA_GRAPH, "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
```

**Review by malfet — APPROVED** [review]

File: test/inductor/test_cudagraph_trees.py [hunk 3]
```diff
@@ -44,6 +45,11 @@
     unittest.skipIf, not HAS_MULTIGPU, "requires multiple cuda devices"
 )
 
+if not TEST_CUDA_GRAPH:
+    if __name__ == "__main__":
+        sys.exit(0)
```
Comment (malfet) [inline test/inductor/test_cudagraph_trees.py:50]: This feels a bit out of place, please add it to https://github.com/pytorch/pytorch/blob/9567aaebe5385c2ce453c10c34180bba53773a93/test/inductor/test_cudagraph_trees.py#L1126-L1130

File: test/dynamo/test_cudagraphs.py [hunk 4]
```diff
@@ -10,7 +10,16 @@
...
-from torch.testing._internal.common_utils import skipIfRocm, TEST_WITH_ROCM
+from torch.testing._internal.common_utils import (
+    skipIfRocm,
+    TEST_CUDA_GRAPH,
+    TEST_WITH_ROCM,
+)
+
+if not TEST_CUDA_GRAPH:
+    if __name__ == "__main__":
+        sys.exit(0)
```
Comment (malfet) [inline test/dynamo/test_cudagraphs.py:21]: Same as the one below, please add it to `if __name__ == "__main__"` at the bottom of the file

File: test/inductor/test_cudagraph_trees.py (hunk 3 above)
Comment (xwang233) [reply test/inductor/test_cudagraph_trees.py]: I was following the syntax just a few lines above it 😅 . I've moved those to the bottom.

**Pull request closed without merging**